# Example_creating_dataset_forRockhound_using_module_features.py_v1

We're hoping that by including a polished dataset that has everything done to it for top prediction except feature engineering and model training, it will be easier to work with and create more experiments out there on this type of problem


### FIND OUT MORE ABOUT ROCKHOUND <a href="https://www.fatiando.org/rockhound/latest/">on their github page</a>

Justin Gosses May 2019
### Uses these modules of Predictatops:
1. features.py

This comes after use of the the wellsKNN.py and several other modules.

This notebook is modeled after the code in features_runner.py in predictatops source folder

First, it should be noted that Predictatops isn't on Pypy, so we're going go to be loading the Python modules from a local directory and this notebook is running from the demo folder in the same Predictatops directory. If you run it from elsewhere, you'll have to change the paths.



In [1]:
ls

Example_firstSteps_modules_fetchdata_configuration_checkdata.ipynb
Example_junk_post_balance.ipynb
Example_module_all_runner_temp_v1b.ipynb
Example_module_balance_v1.ipynb
Example_module_features_v1.ipynb
Example_module_predictionclasses_v1.ipynb
Example_module_trainclasses_v2.ipynb
Example_module_wellsKNN_v1.ipynb
README.md
current_errors_TopMcMr_20181006.png
mannville_demo_data/
mannville_demo_data.zip


In [2]:
cd ..

/Users/justingosses/Code/predictatops


In [3]:
cd predictatops/

/Users/justingosses/Code/predictatops/predictatops


In [4]:
ls

__init__.py                       load_runner.py
__pycache__/                      main.py
all_runner.py                     map.py
balance.py                        plot.py
balance_runner.py                 plot_runner.py
checkdata.py                      predictionclasses.py
checkdata_runner.py               predictionclasses_runner.py
cli.py                            registry.txt
configurationplusfiles.py         split.py
configurationplusfiles_runner.py  split_runner.py
dask-worker-space/                trainclasses.py
features.py                       trainclasses_runner.py
features_runner.py                wellsKNN.py
fetch_demo_data.py                wellsKNN_runner.py
load.py


Now we should be in the Predictatops source file and see various .py files

In [5]:
################ imports ###################
import pandas as pd
################ import from other python files in this package ###################
from features import *
from configurationplusfiles_runner import input_data_inst, config, output_data_inst

/Users/justingosses/anaconda/envs/predictatops/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/Users/justingosses/anaconda/envs/predictatops/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)


head of picks df =      SitID  HorID      Pick  Quality
0  102496   1000       321        1
1  102496   2000                 -1
2  102496   3000                 -1
3  102496   4000                 -1
4  102496   5000       438        2
making base folder for results in: ../results_higherBlanceAndLowerWindows_20190517
base_path directory already exists, ../results_higherBlanceAndLowerWindows_20190517  so not creating it again. This may or may not be what you intended, so just flagging it.
directory  checkData  already exists so not making it again in make_all_directories function of configurationplusfiles.py
directory  load  already exists so not making it again in make_all_directories function of configurationplusfiles.py
directory  split  already exists so not making it again in make_all_directories function of configurationplusfiles.py
directory  wellsKNN  already exists so not making it again in make_all_directories function of configurationplusfiles.py
directory  features  already 

## Code

In [6]:
#########################################
################ code ###################


################ Load dataframe from wellsKNN script ###################

# path_to_prev_results = output_data_inst.base_path_for_all_results
# path_to_directory = output_data_inst.path_wellsKNN
# file_name = output_data_inst.wellsKNN_results_wells_df
# ending = output_data_inst.default_results_file_format

# full_path_to_results_file_from_last_step = getMainDFsavedInStep(path_to_prev_results,path_to_directory,file_name,ending)

# wells_df_from_wellsKNN  = pd.read_hdf(full_path_to_wellsKNN_results)

#### full thing in one step
wells_df_from_wellsKNN = get_wellsKNN_results(output_data_inst)
print("columns",wells_df_from_wellsKNN.columns)
print("len(wells_df_from_wellsKNN 1)",len(wells_df_from_wellsKNN))

columns Index(['SitID', 'lat', 'lng', 'UWI', 'TopHelper_HorID', 'TopTarget_HorID',
       'TopHelper_DEPTH', 'TopTarget_DEPTH', 'TopHelper_HorID_Qual',
       'TopTarget_Qual', 'Neighbors_Obj', 'NN1_topTarget_DEPTH',
       'NN1_TopHelper_DEPTH', 'NN1_thickness',
       'topTarget_Depth_predBy_NN1thick',
       'diff_Top_Depth_Real_v_predBy_NN1thick'],
      dtype='object')
len(wells_df_from_wellsKNN 1) 1601


In [7]:
wells_df_from_wellsKNN.head()

SitID        lat         lng                  UWI  TopHelper_HorID  \
0  102496  54.785907 -110.129320  00/12-08-067-01W4/0            14000   
1  102497  54.782284 -110.269446  00/07-08-067-02W4/0            14000   
2  102498  54.785892 -110.186851  00/09-11-067-02W4/0            14000   
3  102500  54.829624 -110.269422  00/10-29-067-02W4/0            14000   
5  102503  54.771449 -110.402983  00/11-04-067-03W4/0            14000   

   TopTarget_HorID  TopHelper_DEPTH  TopTarget_DEPTH  TopHelper_HorID_Qual  \
0            13000            561.0            475.0                     1   
1            13000            604.5            515.0                     1   
2            13000            564.0            480.0                     1   
3            13000            636.5            549.0                     1   
5            13000            553.5            488.5                     1   

   TopTarget_Qual                                      Neighbors_Obj  \
0               3  [{'neighbor': 1, 'UWI': '00/09-11-067-02W4/0',...   
1               3  [{'neighbor': 1, 'UWI': '00/10-29-067-02W4/0',...   
2               3  [{'neighbor': 1, 'UWI': '00/12-08-067-01W4/0',...   
3               3  [{'neighbor': 1, 'UWI': '00/10-08-068-02W4/0',...   
5               2  [{'neighbor': 1, 'UWI': '00/10-08-067-03W4/0',...   

   NN1_topTarget_DEPTH  NN1_TopHelper_DEPTH  NN1_thickness  \
0                544.0                630.0           86.0   
1                529.0                613.0           84.0   
2                529.0                613.0           84.0   
3                529.0                613.0           84.0   
5                493.5                558.0           64.5   

   topTarget_Depth_predBy_NN1thick  diff_Top_Depth_Real_v_predBy_NN1thick  
0                            475.0                                    0.0  
1                            520.5                                   -5.5  
2                            480.0                                    0.0  
3                            552.5                                   -3.5  
5                            489.0                                   -0.5

In [8]:
################ Load wells df saved from running split functions ###################
################ This file is wells loaded with no features but does have a column for train test split ###################
##### path to input file ######



# split_dir = output_data_inst.base_path_for_all_results+ "/" + output_data_inst.path_split
# split_file = output_data_inst.split_results_wells_df
# ending = output_data_inst.default_results_file_format
# full_path_to_split_results = split_dir+"/"+split_file+ending
# ##### loading wells df created in split ######
# wells_df_from_split  = pd.read_hdf(full_path_to_split_results)

# print("wells_df_from_split.columns",wells_df_from_split.columns)

wells_df_from_split_curveData = get_split_curve_results(output_data_inst)
print("len(wells_df_from_split_curveData)",len(wells_df_from_split_curveData))
print("wells_df_from_split_curveData.columns",wells_df_from_split_curveData.columns)

len(wells_df_from_split_curveData) 1302634
wells_df_from_split_curveData.columns Index(['CALI', 'COND', 'DELT', 'DEPT', 'DPHI', 'DT', 'GR', 'ILD', 'ILM',
       'NPHI', 'PHID', 'RHOB', 'SFL', 'SFLU', 'SN', 'SP', 'UWI',
       'trainOrTest'],
      dtype='object')


In [10]:
wells_df_from_split_curveData.head()

CALI  COND  DELT     DEPT   DPHI  DT       GR    ILD  ILM   NPHI  PHID  \
0  167.003   NaN   NaN  149.602  0.227 NaN  102.473  0.000  NaN  0.460   NaN   
1  199.159   NaN   NaN  149.852  0.263 NaN  122.589  4.202  NaN  0.550   NaN   
2  200.496   NaN   NaN  150.102  0.252 NaN  120.196  4.643  NaN  0.537   NaN   
3  203.933   NaN   NaN  150.352  0.244 NaN  115.975  5.280  NaN  0.513   NaN   
4  203.664   NaN   NaN  150.602  0.240 NaN  109.271  6.592  NaN  0.487   NaN   

   RHOB  SFL  SFLU  SN  SP                      UWI trainOrTest  
0   NaN  NaN   NaN NaN NaN  00-10-32-080-20W4-0.LAS       train  
1   NaN  NaN   NaN NaN NaN  00-10-32-080-20W4-0.LAS       train  
2   NaN  NaN   NaN NaN NaN  00-10-32-080-20W4-0.LAS       train  
3   NaN  NaN   NaN NaN NaN  00-10-32-080-20W4-0.LAS       train  
4   NaN  NaN   NaN NaN NaN  00-10-32-080-20W4-0.LAS       train

In [18]:
wells_df_from_wellsKNN = convertSiteIDListToUWIList(input_data_inst,wells_df_from_wellsKNN)



wells_dict =  {102496: '00-12-08-067-01W4-0.LAS', 102497: '00-07-08-067-02W4-0.LAS', 102498: '00-09-11-067-02W4-0.LAS', 102500: '00-10-29-067-02W4-0.LAS', 102501: '00-06-34-067-02W4-0.LAS', 102503: '00-11-04-067-03W4-0.LAS', 102505: '00-10-08-067-03W4-0.LAS', 102507: '00-10-14-067-03W4-0.LAS', 102514: '00-10-28-067-03W4-0.LAS', 102517: '00-07-36-067-03W4-0.LAS', 102518: '00-10-02-067-04W4-0.LAS', 102565: '00-10-07-067-04W4-0.LAS', 102577: '00-11-24-067-04W4-0.LAS', 102581: '00-07-33-067-04W4-0.LAS', 102583: '00-13-03-067-05W4-0.LAS', 102589: '00-06-16-067-05W4-0.LAS', 102592: '00-07-25-067-05W4-0.LAS', 102596: '00-10-33-067-05W4-0.LAS', 102597: '00-15-01-067-06W4-0.LAS', 102606: '00-06-17-067-06W4-0.LAS', 141727: '00-07-23-067-06W4-0.LAS', 102609: '00-02-29-067-06W4-0.LAS', 102610: '00-11-05-067-07W4-0.LAS', 102611: '00-06-13-067-07W4-0.LAS', 102614: '00-03-25-067-07W4-0.LAS', 102615: '00-08-29-067-07W4-0.LAS', 102619: '00-10-03-067-08W4-0.LAS', 102620: '00-06-13-067-08W4-0.LAS', 14173

In [19]:
print("wells_df_from_wellsKNN.head()",wells_df_from_wellsKNN.head())

wells_df_from_wellsKNN.head()     SitID        lat         lng                      UWI  TopHelper_HorID  \
0  102496  54.785907 -110.129320  00-12-08-067-01W4-0.LAS            14000   
1  102497  54.782284 -110.269446  00-07-08-067-02W4-0.LAS            14000   
2  102498  54.785892 -110.186851  00-09-11-067-02W4-0.LAS            14000   
3  102500  54.829624 -110.269422  00-10-29-067-02W4-0.LAS            14000   
5  102503  54.771449 -110.402983  00-11-04-067-03W4-0.LAS            14000   

   TopTarget_HorID  TopHelper_DEPTH  TopTarget_DEPTH  TopHelper_HorID_Qual  \
0            13000            561.0            475.0                     1   
1            13000            604.5            515.0                     1   
2            13000            564.0            480.0                     1   
3            13000            636.5            549.0                     1   
5            13000            553.5            488.5                     1   

   TopTarget_Qual               

In [20]:
!Which Python

/Users/justingosses/anaconda/envs/predictatops/bin/Python


In [21]:
import sys
sys.executable

'/Users/justingosses/anaconda/envs/predictatops/bin/python'

In [22]:
from platform import python_version
print(python_version())

3.6.6


In [23]:
df_all_wells_wKNN = mergeCurvesAndTopsDF(wells_df_from_split_curveData,wells_df_from_wellsKNN,config)
print("len(df_all_wells_wKNN 2)",len(df_all_wells_wKNN))
print("df_all_wells_wKNN.columns",df_all_wells_wKNN.columns)

print("df_all_wells_wKNN.columns",df_all_wells_wKNN.columns)


len(df_all_wells_wKNN 2) 1302634
df_all_wells_wKNN.columns Index(['CALI', 'COND', 'DELT', 'DEPT', 'DPHI', 'DT', 'GR', 'ILD', 'ILM',
       'NPHI', 'PHID', 'RHOB', 'SFL', 'SFLU', 'SN', 'SP', 'UWI', 'trainOrTest',
       'SitID', 'lat', 'lng', 'TopHelper_HorID', 'TopTarget_HorID',
       'TopHelper_DEPTH', 'TopTarget_DEPTH', 'TopHelper_HorID_Qual',
       'TopTarget_Qual', 'Neighbors_Obj', 'NN1_topTarget_DEPTH',
       'NN1_TopHelper_DEPTH', 'NN1_thickness',
       'topTarget_Depth_predBy_NN1thick',
       'diff_Top_Depth_Real_v_predBy_NN1thick'],
      dtype='object')
df_all_wells_wKNN.columns Index(['CALI', 'COND', 'DELT', 'DEPT', 'DPHI', 'DT', 'GR', 'ILD', 'ILM',
       'NPHI', 'PHID', 'RHOB', 'SFL', 'SFLU', 'SN', 'SP', 'UWI', 'trainOrTest',
       'SitID', 'lat', 'lng', 'TopHelper_HorID', 'TopTarget_HorID',
       'TopHelper_DEPTH', 'TopTarget_DEPTH', 'TopHelper_HorID_Qual',
       'TopTarget_Qual', 'Neighbors_Obj', 'NN1_topTarget_DEPTH',
       'NN1_TopHelper_DEPTH', 'NN1_thickness'

In [24]:
df_all_wells_wKNN.head()

CALI  COND  DELT     DEPT   DPHI  DT       GR    ILD  ILM   NPHI  \
0  167.003   NaN   NaN  149.602  0.227 NaN  102.473  0.000  NaN  0.460   
1  199.159   NaN   NaN  149.852  0.263 NaN  122.589  4.202  NaN  0.550   
2  200.496   NaN   NaN  150.102  0.252 NaN  120.196  4.643  NaN  0.537   
3  203.933   NaN   NaN  150.352  0.244 NaN  115.975  5.280  NaN  0.513   
4  203.664   NaN   NaN  150.602  0.240 NaN  109.271  6.592  NaN  0.487   

                   ...                    TopHelper_DEPTH  TopTarget_DEPTH  \
0                  ...                             384.66           377.95   
1                  ...                             384.66           377.95   
2                  ...                             384.66           377.95   
3                  ...                             384.66           377.95   
4                  ...                             384.66           377.95   

   TopHelper_HorID_Qual  TopTarget_Qual  \
0                     1               3   
1                     1               3   
2                     1               3   
3                     1               3   
4                     1               3   

                                       Neighbors_Obj  NN1_topTarget_DEPTH  \
0  [{'neighbor': 1, 'UWI': '00/10-18-080-20W4/0',...                389.0   
1  [{'neighbor': 1, 'UWI': '00/10-18-080-20W4/0',...                389.0   
2  [{'neighbor': 1, 'UWI': '00/10-18-080-20W4/0',...                389.0   
3  [{'neighbor': 1, 'UWI': '00/10-18-080-20W4/0',...                389.0   
4  [{'neighbor': 1, 'UWI': '00/10-18-080-20W4/0',...                389.0   

  NN1_TopHelper_DEPTH NN1_thickness  topTarget_Depth_predBy_NN1thick  \
0               414.0          25.0                           359.66   
1               414.0          25.0                           359.66   
2               414.0          25.0                           359.66   
3               414.0          25.0                           359.66   
4               414.0          25.0                           359.66   

   diff_Top_Depth_Real_v_predBy_NN1thick  
0                                  18.29  
1                                  18.29  
2                                  18.29  
3                                  18.29  
4                                  18.29  

[5 rows x 33 columns]

In [25]:
#### After joining on the nearest neighbor dataframe, we can cast the original columns to floats instead of strings which some but not necessarily all might be.
#### When we do this, be careful about variation in depth column name and rename DEPTH and DEPT to DEPTH

df_all_wells_wKNN = convertAllColButGivenToFloat(config,df_all_wells_wKNN)

df_all_wells_wKNN = takeLASOffUWI(df_all_wells_wKNN,config)
print("df_all_wells_wKNN.info()",df_all_wells_wKNN.info())
# print("number of columns in len(df_all_wells_wKNN.columns)",len(df_all_wells_wKNN.columns))
# print(df_all_wells_wKNN[0:2])

df_all_wells_wKNN_wEdgesMarked = createDepthRelToKnownTopInSameWell(df_all_wells_wKNN)

df_all_wells_wKNN_wEdgesMarked = createFeat_withinZoneOfKnownPick(df_all_wells_wKNN_wEdgesMarked,config)

df_all_wells_wKNN_wEdgesMarked = NN1_TopMcMDepth_Abs(df_all_wells_wKNN_wEdgesMarked,config)

df_all_wells_wKNN_wEdgesMarked = markingEdgeOfWells(df_all_wells_wKNN_wEdgesMarked,config)

print("df_all_wells_wKNN_wEdgesMarked.head()",df_all_wells_wKNN_wEdgesMarked.head())

turning all columns but these into floats. Should come from config.colsToNotTurnToFloats ['UWI', 'SitID', 'trainOrTest', 'Neighbors_Obj']
<class 'pandas.core.frame.DataFrame'>
Int64Index: 1302634 entries, 0 to 1302633
Data columns (total 33 columns):
CALI                                     447027 non-null float64
COND                                     1550 non-null float64
DELT                                     777 non-null float64
DEPT                                     1302634 non-null float64
DPHI                                     1300552 non-null float64
DT                                       10976 non-null float64
GR                                       1301582 non-null float64
ILD                                      1301802 non-null float64
ILM                                      7037 non-null float64
NPHI                                     1301602 non-null float64
PHID                                     746 non-null float64
RHOB                                    

In [26]:
df_all_wells_wKNN.head()

CALI  COND  DELT     DEPT   DPHI  DT       GR    ILD  ILM   NPHI  \
0  167.003   NaN   NaN  149.602  0.227 NaN  102.473  0.000  NaN  0.460   
1  199.159   NaN   NaN  149.852  0.263 NaN  122.589  4.202  NaN  0.550   
2  200.496   NaN   NaN  150.102  0.252 NaN  120.196  4.643  NaN  0.537   
3  203.933   NaN   NaN  150.352  0.244 NaN  115.975  5.280  NaN  0.513   
4  203.664   NaN   NaN  150.602  0.240 NaN  109.271  6.592  NaN  0.487   

      ...       NN1_thickness  topTarget_Depth_predBy_NN1thick  \
0     ...                25.0                           359.66   
1     ...                25.0                           359.66   
2     ...                25.0                           359.66   
3     ...                25.0                           359.66   
4     ...                25.0                           359.66   

   diff_Top_Depth_Real_v_predBy_NN1thick  diff_TopTarget_DEPTH_v_rowDEPT  \
0                                  18.29                         228.348   
1                                  18.29                         228.098   
2                                  18.29                         227.848   
3                                  18.29                         227.598   
4                                  18.29                         227.348   

   diff_TopHelper_DEPTH_v_rowDEPT  class_DistFrPick_TopTarget  \
0                         235.058                           0   
1                         234.808                           0   
2                         234.558                           0   
3                         234.308                           0   
4                         234.058                           0   

  class_DistFrPick_TopHelper DistFrom_NN1ThickPredTopDepth_toRowDept  NewWell  \
0                          0                                 210.058     True   
1                          0                                 209.808    False   
2                          0                                 209.558    False   
3                          0                                 209.308    False   
4                          0                                 209.058    False   

   LastBitWell  
0        False  
1        False  
2        False  
3        False  
4        False  

[5 rows x 40 columns]

In [27]:
len(df_all_wells_wKNN)

1302634

In [29]:
df_all_wells_wKNN.columns

Index(['CALI', 'COND', 'DELT', 'DEPT', 'DPHI', 'DT', 'GR', 'ILD', 'ILM',
       'NPHI', 'PHID', 'RHOB', 'SFL', 'SFLU', 'SN', 'SP', 'UWI', 'trainOrTest',
       'SitID', 'lat', 'lng', 'TopHelper_HorID', 'TopTarget_HorID',
       'TopHelper_DEPTH', 'TopTarget_DEPTH', 'TopHelper_HorID_Qual',
       'TopTarget_Qual', 'Neighbors_Obj', 'NN1_topTarget_DEPTH',
       'NN1_TopHelper_DEPTH', 'NN1_thickness',
       'topTarget_Depth_predBy_NN1thick',
       'diff_Top_Depth_Real_v_predBy_NN1thick',
       'diff_TopTarget_DEPTH_v_rowDEPT', 'diff_TopHelper_DEPTH_v_rowDEPT',
       'class_DistFrPick_TopTarget', 'class_DistFrPick_TopHelper',
       'DistFrom_NN1ThickPredTopDepth_toRowDept', 'NewWell', 'LastBitWell'],
      dtype='object')

## df_all_wells_wKNN will be the dataset used for import in ROCKHOUND a python package for Download geophysical models/datasets and load them in Python

### FIND OUT MORE ABOUT ROCKHOUND <a href="https://www.fatiando.org/rockhound/latest/">on their github page</a>

We're hoping that by including a polished dataset that has everything done to it for top prediction except feature engineering and model training, it will be easier to work with and create more experiments out there on this type of problem

## Save dataframe as hdf

In [31]:
load_results_full_file_path = "../rockhound_datasets/mcmurry_top_dataframe.h5"

df_all_wells_wKNN.to_hdf(load_results_full_file_path, key='df', mode='w')

/Users/justingosses/anaconda/envs/predictatops/lib/python3.6/site-packages/pandas/core/generic.py:1996: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block3_values] [items->['UWI', 'trainOrTest', 'Neighbors_Obj', 'class_DistFrPick_TopTarget', 'class_DistFrPick_TopHelper']]

  return pytables.to_hdf(path_or_buf, key, self, **kwargs)


In [32]:
test_df = pd.read_hdf(load_results_full_file_path)


In [33]:
test_df.head()

CALI  COND  DELT     DEPT   DPHI  DT       GR    ILD  ILM   NPHI  \
0  167.003   NaN   NaN  149.602  0.227 NaN  102.473  0.000  NaN  0.460   
1  199.159   NaN   NaN  149.852  0.263 NaN  122.589  4.202  NaN  0.550   
2  200.496   NaN   NaN  150.102  0.252 NaN  120.196  4.643  NaN  0.537   
3  203.933   NaN   NaN  150.352  0.244 NaN  115.975  5.280  NaN  0.513   
4  203.664   NaN   NaN  150.602  0.240 NaN  109.271  6.592  NaN  0.487   

      ...       NN1_thickness  topTarget_Depth_predBy_NN1thick  \
0     ...                25.0                           359.66   
1     ...                25.0                           359.66   
2     ...                25.0                           359.66   
3     ...                25.0                           359.66   
4     ...                25.0                           359.66   

   diff_Top_Depth_Real_v_predBy_NN1thick  diff_TopTarget_DEPTH_v_rowDEPT  \
0                                  18.29                         228.348   
1                                  18.29                         228.098   
2                                  18.29                         227.848   
3                                  18.29                         227.598   
4                                  18.29                         227.348   

   diff_TopHelper_DEPTH_v_rowDEPT  class_DistFrPick_TopTarget  \
0                         235.058                           0   
1                         234.808                           0   
2                         234.558                           0   
3                         234.308                           0   
4                         234.058                           0   

  class_DistFrPick_TopHelper DistFrom_NN1ThickPredTopDepth_toRowDept  NewWell  \
0                          0                                 210.058     True   
1                          0                                 209.808    False   
2                          0                                 209.558    False   
3                          0                                 209.308    False   
4                          0                                 209.058    False   

   LastBitWell  
0        False  
1        False  
2        False  
3        False  
4        False  

[5 rows x 40 columns]

## For reference!

###  The configurationplusfiles_runner.py variables used to create this file from the original Mannville datasets were:


##### import from other modules
from configurationplusfiles import *



input_data_inst = input_data('../data/Mannville_input_data/v0.0.0-alpha/OilSandsDB/PICKS.TXT','\t','../data/SPE_006_originalData/OilSandsDB/Logs/*.LAS')

input_data_inst.set_wells_file_path('../data/Mannville_input_data/v0.0.0-alpha/OilSandsDB/WELLS.TXT','\t')

input_data_inst.set_gis_file_path('../data/Mannville_input_data/v0.0.0-alpha/well_lat_lng.csv',',')

input_data_inst.las_folder_path = '../data/Mannville_input_data/v0.0.0-alpha/OilSandsDB/Logs/'

print("head of picks df = ",input_data_inst.picks_df.head())

#### Initiates an object  from the output data class and then creates all the directories for intermediate and final output files
output_data_inst =output_data()

#### CHANGE base_path_for_all_results, was '../results'
output_data_inst.base_path_for_all_results = '../results_higherBlanceAndLowerWindows_20190517'

output_data_inst.make_all_directories()



config = configuration()

config.set_must_have_curves(['ILD', 'NPHI', 'GR', 'DPHI', 'DEPT'])

config.set_must_have_tops__list([13000,14000])

#### Changed for may 17th
#### this change performed worse than default:
## config.curve_windows_for_rolling_features = [3,5,11,21]
#### default is [5,7,11,21]
config.rebalanceClassZeroMultiplier = 120
config.rebalanceClass95Multiplier = 60


config.get_must_have_tops__list()

config.set_top_name_col_in_picks_df('HorID')

print("config.siteID_col_in_picks_df = ",config.siteID_col_in_picks_df)
print("all config is",vars(config))


## Important details to be included in Rockhound python module that imports this file:

<class 'pandas.core.frame.DataFrame'>

Int64Index: 1302634 entries, 0 to 1302633

Data columns (total 33 columns):

CALI                                     447027 non-null float64

COND                                     1550 non-null float64

DELT                                     777 non-null float64

DEPT                                     1302634 non-null float64

DPHI                                     1300552 non-null float64

DT                                       10976 non-null float64

GR                                       1301582 non-null float64

ILD                                      1301802 non-null float64

ILM                                      7037 non-null float64

NPHI                                     1301602 non-null float64

PHID                                     746 non-null float64

RHOB                                     83856 non-null float64

SFL                                      1209 non-null float64

SFLU                                     7025 non-null float64

SN                                       1366 non-null float64

SP                                       11226 non-null float64

UWI                                      1302634 non-null object

trainOrTest                              1302634 non-null object

SitID                                    1302634 non-null int64

lat                                      1302634 non-null float64

lng                                      1302634 non-null float64

TopHelper_HorID                          1302634 non-null float64

TopTarget_HorID                          1302634 non-null float64

TopHelper_DEPTH                          1302634 non-null float64

TopTarget_DEPTH                          1302634 non-null float64

TopHelper_HorID_Qual                     1302634 non-null float64

TopTarget_Qual                           1302634 non-null float64

Neighbors_Obj                            1302634 non-null object

NN1_topTarget_DEPTH                      1299743 non-null float64

NN1_TopHelper_DEPTH                      1302634 non-null float64

NN1_thickness                            1299743 non-null float64

topTarget_Depth_predBy_NN1thick          1299743 non-null float64

diff_Top_Depth_Real_v_predBy_NN1thick    1299743 non-null float64

dtypes: float64(29), int64(1), object(3)

memory usage: 337.9+ MB
